pkg/kubelet/pod_manager_test.go

```diff
@@ -68,26 +68,32 @@ func TestGetSetPods(t *testing.T) {
 	updates := append(expectedPods, mirrorPod)
 	podManager, _ := newFakePodManager()
 	podManager.SetPods(updates)
+
+	// Tests that all regular pods are recorded corrrectly.
 	actualPods := podManager.GetPods()
-	if !reflect.DeepEqual(expectedPods, actualPods) {
-		t.Errorf("pods are not set correctly; expected %#v, got %#v", expectedPods, actualPods)
-	}
-	actualPod, ok := podManager.mirrorPodByUID[mirrorPod.UID]
-	if !ok {
-		t.Errorf("mirror pod %q is not found in the mirror pod map by UID", mirrorPod.UID)
-	} else if !reflect.DeepEqual(&mirrorPod, actualPod) {
-		t.Errorf("mirror pod is recorded incorrectly. expect: %v, got: %v", mirrorPod, actualPod)
-	}
-	actualPod, ok = podManager.mirrorPodByFullName[GetPodFullName(&mirrorPod)]
-	if !ok {
-		t.Errorf("mirror pod %q is not found in the mirror pod map by full name", GetPodFullName(&mirrorPod))
-	} else if !reflect.DeepEqual(&mirrorPod, actualPod) {
-		t.Errorf("mirror pod is recorded incorrectly. expect: %v, got: %v", mirrorPod, actualPod)
+	for _, expected := range expectedPods {
+		found := false
+		for _, actual := range actualPods {
+			if actual.UID == expected.UID {
+				if !reflect.DeepEqual(&expected, &actual) {
+					t.Errorf("pod was recorded incorrectly. expect: %v, got: %v", expected, actual)
+				}
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("pod %q was not found in the pod manager", expected.UID)
```

Comment:
Sorry, I meant in the error (in case that wasn't clear since I haven't seen your new push)